### PR TITLE
Mldb 1361 join rowname subselect

### PR DESCRIPTION
**Pull request by steadtler**
-Fix issue with accessing columns from an outer Join
-Fix issue with using functions with arguments in Join expression

### CODE REVIEW

File: server/function_contexts.cc [hunk 1]
```diff
@@ -283,7 +283,7 @@ doGetFunctionEntity(const Utf8String & functionName)
 }
 
 Utf8String 
-SqlBindingScope::
```
Comment (guyd) [inline server/function_contexts.cc]: The compiler did not complain!

File: testing/MLDB-1361_join_on_subselect.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        ds.record_row("row_a",[["warp", 8, 0]])
+        ds.commit()
+        
+        ds = mldb.create_dataset({ "id": "sub2", "type": "sparse.mutable" })
+        ds.record_row("row_b",[["warp", 9, 0]])
+        ds.commit()
+
+    def test_select_x_works(self):
+        # mldb.get asserts the result status_code is >= 200 and < 400
+        rez = mldb.get("/v1/query", q="""
```
Comment (guyd) [inline testing/MLDB-1361_join_on_subselect.py]: assertQueryResult would make this a little simpler and would have a much better assert message.

File: testing/MLDB-1361_join_on_subselect.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        
+        ds = mldb.create_dataset({ "id": "sub1", "type": "sparse.mutable" })
+        ds.record_row("row_a",[["warp", 8, 0]])
+        ds.commit()
+        
+        ds = mldb.create_dataset({ "id": "sub2", "type": "sparse.mutable" })
+        ds.record_row("row_b",[["warp", 9, 0]])
+        ds.commit()
+
+    def test_select_x_works(self):
```
Comment (guyd) [inline testing/MLDB-1361_join_on_subselect.py]: May be picking a more representation name would help also in finding what is failing.

**Comment by guyd** [comment]
Not too sure about what the transform part does but otherwise the rest is fine.  +1

File: testing/MLDB-1361_join_on_subselect.py (hunk 2 above)
Comment (steadtler) [inline testing/MLDB-1361_join_on_subselect.py]: I was having issues with assertQueryResult and this was urgent. Will try again tomorrow

**Pull request merged**